dev/devicelab/lib/command/upload_results.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:args/command_runner.dart';
+
+import '../framework/cocoon.dart';
+
+class UploadResultsCommand extends Command<void> {
+  UploadResultsCommand() {
+    argParser.addOption('results-file', help: 'Test results JSON to upload to Cocoon.');
+    argParser.addOption(
+      'service-account-token-file',
+      help: 'Authentication token for uploading results.',
+    );
+    argParser.addOption('test-flaky', help: 'Flag to show whether the test is flaky');
+    argParser.addOption(
+      'git-branch',
+      help: '[Flutter infrastructure] Git branch of the current commit. LUCI\n'
+          'checkouts run in detached HEAD state, so the branch must be passed.',
+    );
+    argParser.addOption('luci-builder', help: '[Flutter infrastructure] Name of the LUCI builder being run on.');
+  }
+
+  @override
+  String get name => 'upload-metrics';
```

Comment:
Leave this name unchanged to avoid unexpected breakage.
Will update the name once validated from both here and recipe side.